### ISSUE TITLE
Enhanced signal visibility distance

### ISSUE THREAD

**Issue by flomonster** [OpenRailAssociation/osrd#4923]
### What is this feature about?

Most of our infra have a signal with a visibility distance set to `400m`. It's a great default value but we should take into account the distance between signals. For example, if you have a line with signals close to each other we shouldn't take into account a signal further than the next one (as below).

![image](https://github.com/osrd-project/osrd/assets/15349030/4053a319-462f-40e8-af8e-7a5a70b13a30)

In other words, the visibility distance cannot take on a value greater than the distance between the signals.

### Why is this feature valuable?

More precise spacing resource generation and better GET.
